client/platform/desktop/sharedUtils.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * utilities shared between renderer and main threads.
+ */
+import globToRegexp from 'glob-to-regexp';
+
+function cleanString(dirty: string) {
+  return dirty.replace(/[^a-z0-9]/gi, '_').toLowerCase();
+}
+
+// https://stackoverflow.com/questions/1349404/generate-random-string-characters-in-javascript
+function makeid(length: number): string {
+  let result = '';
+  const characters = 'abcdefghijklmnopqrstuvwxyz0123456789';
+  const charactersLength = characters.length;
+  for (let i = 0; i < length; i += 1) {
+    result += characters.charAt(Math.floor(Math.random() * charactersLength));
+  }
+  return result;
+}
+
+function filterByGlob(pattern: string, files: string[]) {
+  const patterns = pattern.split(';').map((p) => globToRegexp(p, { flags: 'i' }));
+  return files.filter((val) => patterns.some((re) => re.test(val)));
+}
```

Comment:
List of files for front-end and mediaImport are difference so same regEx can have different results.  But there is a difference between the files you are using this function for on the front-end dialog display and the files that are used for the media conversion glob.  The best way to explain it is with an example:
Input folder:
- 2017.0.png
- 2017.1.png
- 2017.2.tiff
- 2017.3.tiff
- 2015.3.png
- 2016.2.tiff
- 2018.2.tiff

Works well if I do `*.png` or `*.tiff`.
If I happen to do `2017*` to filter by year it shows on the front end that I have 4 files.
When you import it will fail because it does the regEx on a different list of files
The output of `globtoRegexp` is `/^2017.*$/i` noting that 2017 is at the beginning of the line.
Inside of `finalizeMediaImport` the passed in files are different:
- /home/local/KHQ/bryon.lewis/dev/TestImages/TestData/tiff_imagelist/2017.0.png
- /home/local/KHQ/bryon.lewis/dev/TestImages/TestData/tiff_imagelist/2017.1.png
- /home/local/KHQ/bryon.lewis/dev/TestImages/TestData/tiff_imagelist/2017.2.tiff
- /home/local/KHQ/bryon.lewis/dev/TestImages/TestData/tiff_imagelist/2017.3.tiff
- /home/local/KHQ/bryon.lewis/dev/TestImages/TestData/tiff_imagelist/2015.3.png
- /home/local/KHQ/bryon.lewis/dev/TestImages/TestData/tiff_imagelist/2016.2.tiff
- /home/local/KHQ/bryon.lewis/dev/TestImages/TestData/tiff_imagelist/2018.2.tiff

So the Regex returns 0 files after the filter and continues with the dataset like it found items.  This should be modified to pass in the same filelist each time.  In addition related to my other comments we should do a check for a return of 0 from this filter in `finalizeMediaImport` and error out like we do in the `beginMediaImport`